python/ray/experimental/sgd/sgd_worker.py

```diff
@@ -113,7 +115,11 @@ def __init__(self,
             manager_socket = (
                 ray.worker.global_worker.plasma_client.manager_socket_name)
             if not plasma.tf_plasma_op:
+                lock_path = os.path.join(pyarrow.__path__[0], "tensorflow", "compile_op.lock")
+                lock = SoftFileLock(lock_path)
+                lock.acquire()
```

Comment:
Let's make this work with a `with` statement?

If not, use `try/finally`, but I think `with` is better.